### ISSUE TITLE
Error 500 on repo deletion

### ISSUE THREAD

**Issue by romainreuillon** [gogits/gogs#1570]
When I try to delete a repo I get an error 500 with a gogs version I just installed from sources. The log is: 

```
2015/09/03 11:41:31 [...ters/repo/setting.go:162 SettingsPost()] [E] DeleteRepository: RepoPath: user does not exist: [uid: 3, name: ]
2015/09/03 11:45:50 [...ters/repo/setting.go:162 SettingsPost()] [E] DeleteRepository: RepoPath: user does not exist: [uid: 9, name: ]
2015/09/03 11:46:22 [...ters/repo/setting.go:162 SettingsPost()] [E] DeleteRepository: RepoPath: user does not exist: [uid: 9, name: ]
```

(I tried with several users)

Romain

**Comment by Unknwon** [comment]
Thanks your feedback!

Are deleting repo from admin panel or in repository settings?

Which exact version of your Gogs?

**Comment by romainreuillon** [comment]
In repository settings.

How can I check the exact version?

**Comment by Unknwon** [comment]
> How can I check the exact version?

At the bottom of the page.

**Comment by romainreuillon** [comment]
Yes, sorry I didn't see that: Version: 0.6.7.0901 Beta

**Comment by Unknwon** [comment]
No worries. So I tested this version which seems no problem with deletion. And what happens to the users with user ID 3 and 9?

**Comment by romainreuillon** [comment]
User 3 is a ldap user and user 9 is a regular user, also I am using sqlite.

**Comment by romainreuillon** [comment]
... when I try to delete I get a page with a big 500 on it. How can I provide more info.

**Comment by Unknwon** [comment]
I'm not quite sure about LDAP user, but delete a repository for a regular user shouldn't be a problem and I have tested without error.

> ... when I try to delete I get a page with a big 500 on it. How can I provide more info?

The log you have posted is all Gogs has right now.

**Comment by focmb** [comment]
Same issue here. I created a repo for testing. I do not using LDAP. Then i treid to delete this repo via the repo settings. The server error message:

    2015/09/03 14:00:48 [...ters/repo/setting.go:162 SettingsPost()] [E] DeleteRepository: RepoPath: user does not exist: [uid: 1, name: ]

Go Version 1.5
Gogs Version 0.6.7.0901 Beta

**Comment by Unknwon** [comment]
@focmb what is your database?

**Comment by focmb** [comment]
sqlite

**Comment by Unknwon** [comment]
Thanks @focmb , found the issue, have pushed fixed code on `develop`.

**Comment by romainreuillon** [comment]
Great, thx!

**Comment by focmb** [comment]
can I test the changes? I tried:
    git clone --depth=1 -b develop https://github.com/gogits/gogs.git
    cd gogs
    go get -u -tags "sqlite"
    go build -tags "sqlite"

I thougt this gives me the version from the develop branch but Gogs shows me 0.6.7.0901 Beta

**Comment by Unknwon** [comment]
Make sure you didn't execute the wrong bina

**Comment by Unknwon** [comment]
*binary

**Comment by focmb** [comment]
Issue solved in the current development version 0.6.9

**Comment by Unknwon** [comment]
@focmb Thanks your confirmation!
